course6/string_ariel/main.cpp

```diff
@@ -0,0 +1,40 @@
+#include "str.h"
+
+int main()
+{
+    //Constructor
+    Str s1 = "foo";
+    std::cout << s1 << std::endl;
```

Comment:
It's a valid interactive test, but in the end you want it to be automated. 
You can write a simple automated test using ```assert```, or do it properly with Google Test or Boost.Test

Reply:
That’s true, did not have time to do a proper unit testing, and hesitated to add 3rd party libraries that cannot build for other people, at least I could’ve used assert tho

Reply:
You should not hesitate to use Boost.
Also, if you not use any 3rd party lib, but simple ```assert``, then it will be simple and universal. Even easier to write and use than all these couts